src/main/scala/it/pagopa/interop/catalogprocess/api/impl/ProcessApiServiceImpl.scala

```diff
@@ -519,6 +520,20 @@ final case class ProcessApiServiceImpl(
     }
   }
 
+  private def deleteInterfaceOnTechnologyUpdate(newTechnology: EServiceTechnology, catalogItem: CatalogItem)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[Unit] = {
+    if (catalogItem.technology.toApi != newTechnology)
+      for {
+        descriptor <- interfaceCanBeDeleted(catalogItem).toFuture
```

Comment:
not required, these steps are performed only if the eservice is in draft (check [here](https://github.com/pagopa/interop-be-catalog-process/pull/237/files?diff=split&w=1#diff-a35fa7da5839824c3e1def5ab0f3eff38f8eb6cd9533b9ec1fcb5b4616723c12R511))
This could be something like
```suggestion
        descriptor <- catalogItem.descriptors.find(_.state == Draft).toFuture
```